netbench-orchestrator/src/russula_cli.rs

```diff
@@ -1,7 +1,177 @@
+    },
+}
+
+#[tokio::main(flavor = "current_thread")]
+async fn main() {
+    let opt = Opt::from_args();
+
+    let file_appender = tracing_appender::rolling::daily("./target", "russula.log");
+    let (non_blocking, _guard) = tracing_appender::non_blocking(file_appender);
+    tracing_subscriber::fmt()
+        .with_env_filter(EnvFilter::from_default_env())
+        .with_writer(non_blocking)
+        .init();
+
+    debug!("{:?}", opt);
+    println!("{:?}", opt);
+    match &opt.workflow {
+        RussulaWorkflow::NetbenchServerWorker { ctx, russula_port } => {
+            let netbench_ctx = ctx.clone();
+            let russula_port = *russula_port;
+            run_server_worker(opt, netbench_ctx, russula_port).await
+        }
+        RussulaWorkflow::NetbenchClientWorker { ctx, russula_port } => {
+            let netbench_ctx = ctx.clone();
+            let russula_port = *russula_port;
+            run_client_worker(opt, netbench_ctx, russula_port).await
+        }
+        RussulaWorkflow::NetbenchServerCoordinator {
+            russula_worker_addrs,
+        } => {
+            let w = russula_worker_addrs.clone();
+            run_local_server_coordinator(opt, w).await
+        }
+        RussulaWorkflow::NetbenchClientCoordinator {
+            russula_worker_addrs,
+        } => {
+            let w = russula_worker_addrs.clone();
+            run_local_client_coordinator(opt, w).await
+        }
+    };
+
+    println!("cli done");
+}
+
+async fn run_server_worker(opt: Opt, netbench_ctx: netbench::ServerContext, russula_port: u16) {
+    let uuid = uuid::Uuid::new_v4().to_string();
+    let id = format!("{}-{}", uuid, netbench_ctx.trim_driver_name());
+    let workflow = server::WorkerWorkflow::new(id, netbench_ctx);
+    let worker = WorkflowBuilder::new(
+        BTreeSet::from_iter([local_listen_addr(russula_port)]),
+        workflow,
+        opt.poll_delay,
+    );
+    let mut worker = worker.build().await.unwrap();
+    worker.run_till(WorkflowState::Ready).await.unwrap();
+
+    worker.run_till(WorkflowState::Done).await.unwrap();
+}
+
+async fn run_client_worker(opt: Opt, netbench_ctx: netbench::ClientContext, russula_port: u16) {
+    let uuid = uuid::Uuid::new_v4().to_string();
+    let id = format!("{}-{}", uuid, netbench_ctx.trim_driver_name());
+    let workflow = client::WorkerWorkflow::new(id, netbench_ctx);
+    let worker = WorkflowBuilder::new(
+        BTreeSet::from_iter([local_listen_addr(russula_port)]),
+        workflow,
+        opt.poll_delay,
+    );
+    let mut worker = worker.build().await.unwrap();
+    worker.run_till(WorkflowState::Ready).await.unwrap();
+
+    worker.run_till(WorkflowState::Done).await.unwrap();
+}
+
+async fn run_local_server_coordinator(opt: Opt, russula_worker_addrs: Vec<SocketAddr>) {
+    let workflow = server::CoordWorkflow::new();
+    let coord = WorkflowBuilder::new(
+        BTreeSet::from_iter(russula_worker_addrs),
+        workflow,
+        opt.poll_delay,
```

Comment:
It does feel like having a hard coded poll delay might simplify this a bit more, since we wouldn't have to pass the argument down through all of these layers.

Reply:
The poll value directly determines how often we poll the workers. I want this to be flexible since it causes more logs to be generated. For testing we can override this value but russula_cli already has a default of [5 sec](https://github.com/aws/s2n-netbench/pull/30/files/b75598824c8372f5d6db9e60c8e8e83dc86b6550#diff-c5a41c48522d88f6623e8691a5da248be1eb81f43bcae7f0d768c82071bda440R23).